server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -1144,11 +1149,28 @@ private static Symbol allocateBuiltinOrUdfFunction(@Nullable String schema,
             }
             newFunction = new Function(signature, castArguments, boundSignature.getReturnType().createType(), filter);
         } else {
-            if (signature.getKind() != FunctionType.WINDOW && signature.getKind() != FunctionType.AGGREGATE) {
-                throw new IllegalArgumentException(String.format(
-                    Locale.ENGLISH,
-                    "OVER clause was specified, but %s is neither a window nor an aggregate function.",
-                    functionName));
+            if (signature.getKind() != FunctionType.WINDOW) {
+                if (signature.getKind() != FunctionType.AGGREGATE) {
+                    throw new IllegalArgumentException(String.format(
+                        Locale.ENGLISH,
+                        "OVER clause was specified, but %s is neither a window nor an aggregate function.",
+                        functionName));
+                } else {
+                    if (ignoreNulls != null) {
+                        throw new IllegalArgumentException(String.format(
+                            Locale.ENGLISH,
+                            "%s cannot accept RESPECT or IGNORE NULLS flag.",
+                            functionName));
+                    }
+                }
+            } else {
+                // cannot access these static strings as they are in extensions module
+                if (!Set.of("first_value","last_value","nth_value","lead","lag").contains(functionName) && ignoreNulls != null) {
```

Comment:
please suggest how I could avoid hard-coding here..

Reply:
I think it should be up to the function implementation to tell whether it can support `ignoreNulls` or not. 
So we should defer this check until the implementation is resolved.